File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.py
def solution(wallet, bill):
    answer = 0
    
#     max_wallet = max(wallet)
#     min_wallet = min(wallet)
    
#     max_bill = max(bill)
#     min_bill = min(bill)
    
    wallet.sort(reverse=True)
    bill.sort(reverse=True)
    
    while True:
        if not (bill[1] > wallet[1] or bill[0] > wallet[0]):
            break
        bill[0] = bill[0] // 2
        bill.sort(reverse=True)
        answer += 1
        
    
    return answer

File: 프로그래머스/1/340199. ［PCCE 기출문제］ 9번 ／ 지폐 접기/［PCCE 기출문제］ 9번 ／ 지폐 접기.java
import java.util.*;

class Solution {
    public int solution(int[] wallet, int[] bill) {
        int answer = 0;
        
        int minWallet = Math.min(wallet[0], wallet[1]);
        int maxWallet = Math.max(wallet[0], wallet[1]);
        
        while (true) {
            int minBill = Math.min(bill[0], bill[1]);
            int maxBill = Math.max(bill[0], bill[1]);
            
            if (minBill > minWallet || maxBill > maxWallet){
                if (bill[0] > bill[1]){
                    bill[0] /= 2;
                } else{
                    bill[1] /=2;
                }
                answer++;
            } else{
                break;
            }
        }
        
        
        return answer;
    }
}